chore: Add stetho

KKS161994 opened the pull request:
Added stetho as dependency and initialised it in MainApplication.

KKS161994 commented:
Can you please let me know if their are some other changes that I need to do.

iamareebjamal commented on app/src/main/java/org/fossasia/susi/ai/MainApplication.kt:
```diff
@@ -50,6 +53,8 @@ class MainApplication : Application() {
             Timber.plant(ReleaseLogTree())
         }
+        Stetho.initializeWithDefaults(this)
+        OkHttpClient.Builder().addNetworkInterceptor(StethoInterceptor()).build()
```
Won't work. Please ensure it works before sending PR

KKS161994 commented:
Sir I have added NetworkInterceptor, Js module, dumpapp, local storage module of stetho. Is their anything more to add sir.
I have tested it from terminal using ./dumpapp and also using inspector of chrome for network requests and local storage.

iamareebjamal commented on app/build.gradle:
```diff
@@ -181,4 +181,14 @@ dependencies {
+    //Stetho network helper
+    implementation 'com.facebook.stetho:stetho-okhttp3:1.5.1'
+    //Stetho js module
+    implementation 'com.facebook.stetho:stetho-js-rhino:1.4.2'
```
Should only be enabled in debug build

KKS161994 commented:
Sir I have added it as a debug implementation and also the no-op part for stetho so that it doesn't fail for release build too.
releaseCompile "net.igenius:stetho-no-op:1.1"
Above is no-op library for stetho .

iamareebjamal commented on app/src/main/java/org/fossasia/susi/ai/MainApplication.kt:
```diff
@@ -50,6 +51,12 @@ class MainApplication : Application() {
+                Stetho.newInitializerBuilder(this)
+                        .enableDumpapp(Stetho.defaultDumperPluginsProvider(this))
+                        .enableWebKitInspector(Stetho.defaultInspectorModulesProvider(this))
+                        .build())
```
Use initializeWithDefaults

KKS161994 commented:
Sir to use Stetho.initialisewithdefaults() I have to use a different no-op library since the older one didn't have initialiseWithdefaults() as a  member method. The new library has it included with other methods too.

iamareebjamal commented:
https://github.com/iamareebjamal/stetho-noop

<blockquote><img src="https://avatars1.githubusercontent.com/u/3874064?s=400&v=4" width="48" align="right"><div><img src="https://github.githubassets.com/favicon.ico" height="14"> GitHub</div><div><strong><a href="https://github.com/iamareebjamal/stetho-noop">iamareebjamal/stetho-noop</a></strong></div><div>Facebook stetho no-op for release build. Contribute to iamareebjamal/stetho-noop development by creating an account on GitHub.</div></blockquote>

KKS161994 commented:
> https://github.com/iamareebjamal/stetho-noop
> 
> > <img alt="" width="48" src="https://avatars1.githubusercontent.com/u/3874064?s=400&amp;v=4"><img alt="" height="14" src="https://camo.githubusercontent.com/4908d9947135e4536ef61d61512ee575f869bf6f/68747470733a2f2f6769746875622e6769746875626173736574732e636f6d2f66617669636f6e2e69636f"> GitHub**[iamareebjamal/stetho-noop](https://github.com/iamareebjamal/stetho-noop)**Facebook stetho no-op for release build. Contribute to iamareebjamal/stetho-noop development by creating an account on GitHub.

Sir I integrated it for release implementation .

<blockquote><img src="https://avatars1.githubusercontent.com/u/3874064?s=400&v=4" width="48" align="right"><div><img src="https://github.githubassets.com/favicon.ico" height="14"> GitHub</div><div><strong><a href="https://github.com/iamareebjamal/stetho-noop">iamareebjamal/stetho-noop</a></strong></div><div>Facebook stetho no-op for release build. Contribute to iamareebjamal/stetho-noop development by creating an account on GitHub.</div></blockquote>

iamareebjamal commented on app/build.gradle:
```diff
@@ -181,4 +181,17 @@ dependencies {
+    debugImplementation "com.facebook.stetho:stetho-okhttp3:${rootConfiguration.stethoVersion}"
+
+    //Stetho js module
+    debugImplementation "com.facebook.stetho:stetho-js-rhino:${rootConfiguration.stehtoJsRhinoVersion}"
```
This isn't required

iamareebjamal commented on app/config.gradle:
```diff
@@ -25,4 +25,7 @@ ext {
     mockitoCoreVersion = '1.10.19'
     playServiceAuthVersion='16.0.1'
     shimmerVersion = "0.5.0"
+    jitpackVersion = "1.1"
```
There is no library named 'jitpack'

Also, this section id only for dependencies with duplicate versions

KKS161994 commented:
Sir I have removed the js module of Stetho. And also the version numbers that I added. Okhttp dependency is a must to have for monitoring network requests made.

iamareebjamal commented on app/build.gradle:
```diff
@@ -181,4 +181,12 @@ dependencies {
+    debugImplementation "com.facebook.stetho:stetho:1.5.1"
+
+    //Stetho network helper
+    debugImplementation "com.facebook.stetho:stetho-okhttp3:1.5.1"
```
This version is duplicate. Add it to the versions

KKS161994 commented:
Created a version name in the versions for Stetho and stetho ok http sir.
